python/lambda-cloudwatch-dashboard/lambda_cloudwatch_dashboard/lambda_cloudwatch_dashboard_stack.py

```diff
@@ -18,13 +18,14 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
                                     runtime=aws_lambda.Runtime.PYTHON_3_7,
                                     handler="lambda-handler.main",
                                     code=aws_lambda.Code.from_asset("./lambda"))
-        lambda_function.role.add_managed_policy(
-            aws_iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AWSLambdaBasicExecutionRole"))
+        if lambda_function.role is not None:
+            lambda_function.role.add_managed_policy(
+                aws_iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AWSLambdaBasicExecutionRole"))
 
-        # Create CloudWatch Dashboard to view Lambda Function Metrics
-        cw_dashboard = aws_cloudwatch.Dashboard(self, "Lambda Dashboard",
-          dashboard_name=example_dashboard_name
-        )
+            # Create CloudWatch Dashboard to view Lambda Function Metrics
```

Comment:
Introduced bug: If the lambda function role is None, the cw_dashboard is not created and will cause the [line below](https://github.com/aws-samples/aws-cdk-examples/pull/554/files#diff-7e3a0b4815c719a61b01cb018703c2181e732b2da024182870100a7b8ca378f4R57) to fail.